src/prefect/tasks/mysql/mysql.py

```diff
@@ -46,32 +46,58 @@ def __init__(
         self.charset = charset
         super().__init__(**kwargs)
 
-    @defaults_from_attrs("query", "commit", "charset")
-    def run(self, query: str, commit: bool = False, charset: str = "utf8mb4") -> int:
+    @defaults_from_attrs(
+        "db_name", "user", "password", "host", "port", "query", "commit", "charset"
+    )
+    def run(
+        self,
+        db_name: str = None,
+        user: str = None,
+        password: str = None,
+        host: str = None,
+        port: int = None,
+        query: str = None,
+        commit: bool = None,
+        charset: str = None,
+    ) -> int:
         """
         Task run method. Executes a query against MySQL database.
 
         Args:
+            - db_name (str): name of MySQL database
+            - user (str): user name used to authenticate
+            - password (str): password used to authenticate
+            - host (str): database host address
+            - port (int, optional): port used to connect to MySQL database, defaults to 3307
+                if not provided
             - query (str, optional): query to execute against database
-            - commit (bool, optional): set to True to commit transaction, defaults to False
-            - charset (str, optional): charset of the query, defaults to "utf8mb4"
+            - commit (bool, optional): set to True to commit transaction, defaults to false
+            - charset (str, optional): charset you want to use (defaults to utf8mb4)
 
         Returns:
             - executed (int): number of affected rows
 
         Raises:
-            - pymysql.MySQLError
+            - pymysql.MySQLError: if exception occurs when executing the query
         """
+        if not db_name:
+            raise ValueError("A db_name must be provided")
+        if not user:
+            raise ValueError("A user must be provided")
+        if not password:
+            raise ValueError("A password must be provided")
+        if not host:
+            raise ValueError("A host must be provided")
         if not query:
             raise ValueError("A query string must be provided")
 
         conn = pymysql.connect(
-            host=self.host,
-            user=self.user,
-            password=self.password,
-            db=self.db_name,
-            charset=self.charset,
-            port=self.port,
+            host=(host or self.host),
```

Comment:
the decorator on this line takes care of this logic for you https://github.com/PrefectHQ/prefect/pull/4363/files#diff-dcbaaa9d19c03a04f422b74f5edbc0a28717268dd9b4492320337d74b082bafcR49